src_c/image.c

```diff
@@ -314,6 +314,129 @@ image_get_extended(PyObject *self, PyObject *arg)
     return PyInt_FromLong(GETSTATE(self)->is_extended);
 }
 
+static PG_INLINE int
+unpack_RGB_color(char *data, Uint32 color, SDL_PixelFormat *format) {
+    data[0] = (char)(((color & format->Rmask) >> format->Rshift) << format->Rloss);
+    data[1] = (char)(((color & format->Gmask) >> format->Gshift) << format->Gloss);
+    data[2] = (char)(((color & format->Bmask) >> format->Bshift) << format->Bloss);
+    return 3;
+}
+
+static PG_INLINE int
+unpack_RGB_color_alpha_mult(char *data, Uint32 color, SDL_PixelFormat *format, Uint32 alpha) {
+    data[0] = (char)(
+                      (((color & format->Rmask) >> format->Rshift) << format->Rloss) *
+                      alpha / 255
+                  );
+    data[1] = (char)(
+                      (((color & format->Gmask) >> format->Gshift) << format->Gloss) *
+                      alpha / 255
+                  );
+    data[2] = (char)(
+                      (((color & format->Bmask) >> format->Bshift) << format->Bloss) *
+                      alpha / 255
+                  );
+    return 3;
+}
+
+static PG_INLINE int
+unpack_RGBA_color_alpha_mult(char *data, Uint32 color, SDL_PixelFormat *format) {
+    Uint32 alpha = ((color & format->Amask) >> format->Ashift) << format->Aloss;
+    data[3] = (char)alpha;
+    return 1 + unpack_RGB_color_alpha_mult(data, color, format, alpha);
+}
+
+static PG_INLINE int
+unpack_ARGB_color_alpha_mult(char *data, Uint32 color, SDL_PixelFormat *format) {
+    Uint32 alpha = ((color & format->Amask) >> format->Ashift) << format->Aloss;
+    data[0] = (char)alpha;
+    return 1 + unpack_RGB_color_alpha_mult(data + 1, color, format, alpha);
+}
+
+
+static PG_INLINE int
+unpack_ARGB_color(char *data, Uint32 color, SDL_PixelFormat *format) {
+    if (format->Amask) {
+        data[0] = (char)(((color & format->Amask) >> format->Ashift) << format->Aloss);
+    } else {
+        data[0] = (char)255;
+    }
+    return 1 + unpack_RGB_color(data + 1, color, format);
+}
+
+
+static PG_INLINE int
+unpack_color_from_palette(char *data, Uint32 color, SDL_PixelFormat *format) {
+    data[0] = (char)format->palette->colors[color].r;
+    data[1] = (char)format->palette->colors[color].g;
+    data[2] = (char)format->palette->colors[color].b;
+    return 3;
+}
+
+static PG_INLINE Uint32 read_24bit_color(const Uint8 *ptr) {
+#if SDL_BYTEORDER == SDL_LIL_ENDIAN
+    return ptr[0] | (ptr[1] << 8) | (ptr[2] << 16);
+#else
+    return ptr[2] | (ptr[1] << 8) | (ptr[0] << 16);
+#endif
+}
+
+typedef int (*unpack_function)(char *, Uint32, SDL_PixelFormat *);
+
+static PG_INLINE void
+parse_8bit_colors(SDL_Surface *surf, char *data, int flipped, unpack_function unpack_func) {
```

Comment:
I'm not sure about `char *data`.

- data could be a different name? `dataout`?
- the `char *` could be `Uint8 *`?